### PR TITLE
Wifi configuration in EEPROM

### PR DESCRIPTION
**Pull request by vorner**
This adds support for:

* Loading the configuration from INI file by a menu item (very minimalistic, error reporting and success indication is still needed and applying the values right away).
* Storing the configuration in EEPROM.
* Adding some type safety improvements around EEPROM.

BFW-2110

### CODE REVIEW

File: src/common/eeprom.h [hunk 1]
```diff
@@ -21,7 +21,7 @@ enum {
     MAX_SHEET_NAME_LENGTH = 8,
 };
 
-enum {
+enum eevar_id {
```
Comment (marekmosna-prusa3d) [inline src/common/eeprom.h:24]: please revert such a change. It doesn't relate to given task

**Comment by marekmosna-prusa3d** [comment]
Please try to focus only on very core of the feature.  Improvements will follow as soon as we plan and discuss them with the rest of the team.

File: src/gui/screen_menu_lan_settings.cpp [hunk 2]
```diff
@@ -65,13 +65,26 @@ class MI_NET_IP_t : public WI_SWITCH_t<2> {
...
+        // FIXME: Some error reporting
+        // FIXME: How do we _activate_ the new settings?
+        net_load_ini_to_eeprom();
+    }
+};
+
 /*****************************************************************************/
 //parent alias
-using MenuContainer = WinMenuContainer<MI_RETURN, MI_NET_INTERFACE_t, MI_NET_IP_t>;
+using MenuContainer = WinMenuContainer<MI_RETURN, MI_NET_INTERFACE_t, MI_NET_IP_t, MI_LAN_LOAD_t>;
```
[the remark this answers is not in the document]
Comment (vorner) [reply src/gui/screen_menu_lan_settings.cpp:83]: Are you sure? It does load only network related changes and originally it _was_ in LAN settings. Is there a guideline or other reason why you propose putting it one level to the main menu?

**Comment by vorner** [comment]
OK, the refactoring commit is now officially part of #1653, let's wait for it to get merged first so it doesn't show here (the trick of assigning that branch as the target branch doesn't work with this way of fork management :-| ).

**Review by marekmosna-prusa3d — APPROVED** [review]

File: lib/WUI/wui_api.c [hunk 3]
```diff
@@ -154,71 +158,136 @@ uint32_t load_ini_file_wifi(ETH_config_t *config, ap_entry_t *ap) {
...
+    default:
+        assert(0 /* Unknown net device */);
+    }
 
-    if (ethconfig->var_mask & ETHVAR_MSK(ETHVAR_LAN_FLAGS)) {
-        eeprom_set_var(EEVAR_LAN_FLAG, variant8_ui8(ethconfig->lan.flag));
+    return id + offset;
+}
+
+#define ID(VAR_ID) vid((VAR_ID), netdev_id)
```
[the remark this answers is not in the document]
Comment (vorner) [reply lib/WUI/wui_api.c]: Unfortunately, I used the macro in two functions. But OK, I can use the above function directly (there would be no use of macro any more with added parameter).

**Comment by vorner** [comment]
Thank you :-)

**Pull request merged**